files/ebs_volume.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+until ping -c 1 google.com &> /dev/null; do
+  echo "waiting for outbound connectivity"
+  sleep 5
+done
+
+systemctl stop graphdb
+
+# Set common variables used throughout the script.
+imds_token=$( curl -Ss -H "X-aws-ec2-metadata-token-ttl-seconds: 300" -XPUT 169.254.169.254/latest/api/token )
+instance_id=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/instance-id )
+availability_zone=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/placement/availability-zone )
+volume_id=""
+
+# Search for an available EBS volume to attach to the instance. Wait one minute for a volume to become available,
+# if no volume is found - create new one, attach, format and mount the volume.
+
+for i in $(seq 1 6); do
+
+  volume_id=$(
+    aws --cli-connect-timeout 300 ec2 describe-volumes \
+      --filters "Name=status,Values=available" "Name=availability-zone,Values=$availability_zone" "Name=tag:Name,Values=${name}-graphdb-data" \
+      --query "Volumes[*].{ID:VolumeId}" \
+      --output text | \
+      sed '/^$/d'
+  )
+
+  if [ -z "$${volume_id:-}" ]; then
+    echo 'ebs volume not yet available'
+    sleep 10
+  else
+    break
+  fi
+done
+
+if [ -z "$${volume_id:-}" ]; then
+
+# if volume type is gp3, then we don't need to give value to "throughput" param
```

Comment:
IO disks don't require throughput, gp3 do.

Reply:
I have reversed the logic. Thanks!